lib/elixir/lib/kernel/utils.ex

```diff
@@ -22,27 +22,41 @@ defmodule Kernel.Utils do
         _ -> raise ArgumentError, "invalid syntax in defdelegate #{Macro.to_string(fun)}"
       end
 
-    :ok = check_defdelegate_args(args)
-
-    as_args =
-      case append_first and args != [] do
-        true  -> tl(args) ++ [hd(args)]
-        false -> args
+    as_args_list = normalize_args(args)
+    as = Keyword.get(opts, :as, name)
+    Enum.map as_args_list, fn as_args ->
+      formal_args = make_formal_args(as_args)
+      as_args = case append_first do
+        true  -> tl(as_args) ++ [hd(as_args)]
+        false -> as_args
       end
+      {name, formal_args, as, as_args}
+    end
+  end
 
-    as = Keyword.get(opts, :as, name)
-    {name, args, as, as_args}
+  defp make_formal_args(args) do
+    Enum.filter args, fn
+      {name, _, mod} when is_atom(name) and is_atom(mod) -> true
+      _                                                  -> false
+    end
   end
 
-  defp check_defdelegate_args([]),
-    do: :ok
-  defp check_defdelegate_args([{var, _, mod}|rest]) when is_atom(var) and is_atom(mod),
-    do: check_defdelegate_args(rest)
-  defp check_defdelegate_args([code|_]) do
-    raise ArgumentError,
-      "defdelegate/2 only accepts variable names, got: #{Macro.to_string(code)}"
+  defp normalize_args(raw_args) do
+    List.foldr raw_args, [[]], fn
+      ({:\\, _, [arg, default_arg]}, [as_args|_] = as_args_list) ->
```

Comment:
You need to check `arg` here is also a variable. I would suggesting move the checking logic and the error to `add_arg`.